activerecord/lib/active_record/persistence.rb

```diff
@@ -623,14 +617,17 @@ def discriminate_class_for_record(record)
           self
         end
 
-        # Called by +_update_record+ and +_delete_record+
-        # to build `where` clause from default scopes.
+        # Called by +_update_record+ and +_delete_record+ to build `where`
+        # clause from scopes, default or from +scoping+
         # Skips empty scopes.
-        def build_default_constraint
-          return unless default_scopes?(all_queries: true)
+        def build_scope_constraints
+          scope_constraints = if current_scope = global_current_scope
```

Comment:
We are probably missing a test case when both default scope and a `global_current_scope` are present. We used to apply both of them while now we will only apply one of each with `global_current_scope` having a priority.

So a code like

```ruby
    Comment.where(likes_count: (10..100)).scoping do
       Comment.first
    end
```
will change the behavior by not applying the default scope. (I haven't actually tested it yet)
Any chance that it's actually the desired behavior of `.scoping` ? 

So as you mentioned in the PR description
> but this is the only way I found to detect that .uncoped was called earlier and to prevent applying the default scopes to the current operation.

we may actually need to find a clear way of telling that the `global_current_scope` is the `.unscoped` one.
Almost like `return if global_current_scope.unscoped?` which would translate as "do not apply anything if we are inside an unscoped context"

Reply:
Totally, and maybe only tangentially related, but [this issue](https://github.com/rails/rails/issues/47357) highlights similar issues in terms of how and when scopes are applied.

> we may actually need to find a clear way of telling that the global_current_scope is the .unscoped one.
Almost like return if global_current_scope.unscoped? which would translate as "do not apply anything if we are inside an unscoped context"

I think that it would be great. I haven't spent enough time in the code, and am not familiar enough to know how easy or not it would be to do that, but it sounds like a good direction.